design/setup.md

```diff
@@ -127,3 +124,23 @@ This is determined through the [DC/OS variant](https://jira.mesosphere.com/brows
 (new in 1.12). For previous versions of DC/OS we won’t try to detect open / enterprise as it’d involve
 some hacks, but rather display a message saying “Please run “dcos package install dcos-enterprise-cli” if
 you use a DC/OS Enterprise cluster”.
+
+### Installation mechanism
+
+The CLI will first try to auto-install a plugin using its canonical URL:
+
+- https://downloads.dcos.io/cli/{stability}/plugins/{plugin}/{platform}/x86-64/{plugin}-{dcos-version}-patch.x.zip
+
+The URL placeholders are defined as below:
+
+- `plugin` is either `dcos-core-cli` or `dcos-enterprise-cli`
+- `stability` is set to `testing` when the CLI is a dev build, otherwise defaults to `releases`
+- `platform` can either be `linux`, `darwin`, or `windows`
+- `dcos-version` is the major and minor version of the DC/OS cluster (eg. `1.13`)
+
+Failing to download the plugins from their canonical URL usually means that the CLI user operates from an
+air-gapped environment. In that case the CLI falls back to downloading the plugins through Cosmos, where they are registered as packages named `dcos-core-cli` and `dcos-enterprise-cli`.
+
+Failing to download the plugins from Cosmos usually means that the CLI user doesn't have the necessary
+permissions in order to interact with Cosmos. In that case the setup command fails with an error message
+indicating a web-page the user can go to in order to download the plugins manually.
```

Comment:
Is it sufficient enough to just tell users the location of the plugins? Do they know how to add them?
Otherwise LGTM.